prusti-viper/src/encoder/errors/error_manager.rs

```diff
@@ -69,6 +69,9 @@ pub enum ErrorCtxt {
     AssertTerminator(String),
     /// A Viper `assert false` in the context of a bounds check
     BoundsCheckAssert,
+    /// A Viper `assert false` in the context of a hardcoded bounds check (e.g. when we hardcode a `index`)
+    /// TODO: remove this in favor of extern_spec for e.g. the stdlib `fn index(...)`
+    HardcodedBoundsCheckAssert(String),
```

Comment:
What about calling this variant `SliceRangeBoundsCheckAssert`? It took me a minute to understand the meaning of hardcoded in this context.